core/blockchain.go

```diff
@@ -785,15 +781,18 @@ func (bc *BlockChain) setHeadBeyondRoot(head uint64, time uint64, root common.Ha
 			// last step, however the direction of SetHead is from high
 			// to low, so it's safe to update in-memory markers directly.
 			bc.currentBlock.Store(newHeadBlock.Header())
-			headBlockGauge.Update(int64(newHeadBlock.NumberU64()))
+			headBlockGauge.Update(newHeadBlock.Number().Int64())
```

Comment:
It's not trivial - but we need to take changes we made to SetHeadBeyondRoot and apply them to rewindPathHead and rewindHashHead, calling the original rewindHead (from v1.14.0) from setHeadBeyondRoot